File: src/Software Engineer Prepare Kit/01/solution.go
package main

func CountResponseTimeRegressions(responseTimes []int32) int32 {
	if len(responseTimes) < 2 {
		return 0
	}

	var prevTimesTotal int64 = int64(responseTimes[0])
	var timesGreaterThanAverage int32

	for i := 1; i < len(responseTimes); i++ {
		time := responseTimes[i]
		prevTimesAverage := float64(prevTimesTotal) / float64(i)

		if float64(time) > prevTimesAverage {
			timesGreaterThanAverage++
		}

		prevTimesTotal += int64(time)
	}

	return timesGreaterThanAverage
}

File: src/Software Engineer Prepare Kit/01/solution.ts
export default function countResponseTimeRegressions(responseTimes: number[]): number {
	let prevTimesTotal = responseTimes[0]
	let timesGreaterThanAverage = 0

	for (let i = 1; i < responseTimes.length; i++) {
		const time = responseTimes[i]!
		const prevTimesAverage = prevTimesTotal! / i

		timesGreaterThanAverage += ~~(time > prevTimesAverage)
		prevTimesTotal! += time
	}

	return timesGreaterThanAverage
}
